crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -259,30 +259,24 @@ impl AssetLoader for ShaderLoader {
     ) -> BoxedFuture<'a, Result<Shader, Self::Error>> {
         Box::pin(async move {
             let ext = load_context.path().extension().unwrap().to_str().unwrap();
-
+            let path = load_context.asset_path().to_string();
+            // On windows, the path will inconsistently use \ or /.
+            // TODO: remove this once AssetPath forces cross-platform "slash" consistency. See #10511
+            let path = path.replace("\\", "/");
```

Comment:
```suggestion
            let path = path.replace(std::path::MAIN_SEPARATOR, "/");
```
should work if someday there's a platform with another kind of separator...